GVFS/GVFS.Platform.Mac/MacFileSystem.cs

```diff
@@ -56,7 +56,9 @@ private class NativeFileReader
             public static bool TryReadFirstByteOfFile(string fileName, byte[] buffer)
             {
                 int fileDescriptor = Open(fileName, ReadOnly);
-                return TryReadOneByte(fileDescriptor, buffer);
+                bool readStatus = TryReadOneByte(fileDescriptor, buffer);
+                Close(fileDescriptor);

```

Comment:
What would you think of putting the `Close` inside of a `finally`?

`TryReadOneByte` shouldn't ever throw, but using a `try`\ `finally` gets us the same behavior as a `using` statement.